### DISCUSSION TITLE
Dynamic route-separated ApolloServer instances

### DISCUSSION THREAD

**Discussion by mgagliardo91** [apollographql/apollo-server#4391] [General]
I am working on a project where I would like to schemas separated by organization and host an organization-specific ApolloServer instance at a route prefixed by the organization name.

For example, the route `/api/:organizationId/graphql` would surface an ApolloServer instance with a schema specific to the organization matching `:organizationId`. I am currently using apollo-server-express, however, it requires having a route to apply its middleware ahead of time, rather than dynamically.

Ultimately, I would imagine an architecture where when a request comes in for a specific organization, I would initialize an ApolloServer instance with the org-specific schema, cache it, and reuse it for subsequent requests to the same org-prefixed route.

Has anyone done anything similar to this and, if so, how would I best implement this with ApolloServer in node/express?

Thanks!

**Comment by cachila** [comment]
Hi @mgagliardo91,

Did you find any solution to this?
